src/analysis/syntax_doc.ml

```diff
@@ -0,0 +1,286 @@
+      }
+  | (_, Constructor_declaration _)
+    :: (_, Type_kind (Ttype_variant _))
+    :: (_, Type_declaration { typ_private = Public; _ })
+    :: _ ->
+      {
+        name = "Variant Types";
+        description =
+          "Let's you represent data that may take on multiple different forms.";
+        documentation =
+          "https://v2.ocaml.org/manual/coreexamples.html#s:tut-recvariants";
+      }
+  | _
+    :: (_, Constructor_declaration _)
+    :: (_, Type_kind (Ttype_variant _))
+    :: (_, Type_declaration { typ_private = Public; _ })
+    :: _ ->
+      {
+        name = "Variant Types";
+        description =
+          "Let's you represent data that may take on multiple different forms.";
+        documentation =
+          "https://v2.ocaml.org/manual/coreexamples.html#s:tut-recvariants";
+      }
+  | (_, Type_kind Ttype_open)
+    :: (_, Type_declaration { typ_private = Public; _ })
+    :: _ ->
+      {
+        name = "Extensible variant types";
+        description = "Can be extended with new variant constructors using +=.";
+        documentation = "https://v2.ocaml.org/manual/extensiblevariants.html";
+      }
+  | (_, Type_kind Ttype_abstract)
+    :: (_, Type_declaration { typ_private = Public; _ })
+    :: _ ->
+      {
+        name = "Abstract types";
+        description =
+          "Allows you to define variants with arbitrary data structures, \
+           including other variants, records, and functions";
+        documentation = "https://v2.ocaml.org/manual/extensiblevariants.html";
+      }
+  | (_, Core_type _)
+    :: (_, Core_type _)
+    :: (_, Label_declaration _)
+    :: (_, Type_kind (Ttype_record _))
+    :: (_, Type_declaration { typ_private = Public; _ })
+    :: _ ->
+      {
+        name = "Record types";
+        description =
+          "Allows you to define variants with a fixed set of fields, and all \
+           of the constructors for a record variant type must have the same \
+           fields";
+        documentation = "https://v2.ocaml.org/manual/extensiblevariants.html";
+      }
+  | (_, Type_kind _) :: (_, Type_declaration { typ_private = Public; _ }) :: _
+    ->
+      {
+        name = "Empty Variant types";
+        description = "This extension allows the user to define empty variants.";
+        documentation = "https://v2.ocaml.org/manual/emptyvariants.html";
+      }
+  | (_, Constructor_declaration _)
+    :: (_, Type_kind (Ttype_variant _))
+    :: (_, Type_declaration { typ_private = Private; _ })
+    :: _ ->
+      {
+        name = "Private Types";
+        description =
+          "Values of a variant type declared private can be de-structured \
+           normally in pattern-matching. However, values of these types cannot \
+           be constructed directly by constructor application.";
+        documentation =
+          "https://v2.ocaml.org/releases/5.1/htmlman/privatetypes.html#ss:private-types-variant";
```

Comment:
Some urls start with `https://v2.ocaml.org/releases/5.1/htmlman` and others with `https://v2.ocaml.org/manual/`.
Is that expected ?

I guess we should always point to same version of the manual and `https://v2.ocaml.org/releases/5.1/htmlman/` are the more precise ones ?

You could make a small builder function that takes only the changing part of the url (like `privatetypes.html#ss:private-types-variant
`) and concatenates it to the base url. This way it would easier to change the manual version for other Merlin branches.

And in fact, you are targeting Merlin for OCaml 4.14 here, so you should point to the 4.14's manual 👍

Reply:
Thank you. This is a brilliant idea.